### PR TITLE
Update Leads - Basic Info

### PR DESCRIPTION
**Pull request by MeghanBomberger**
Part of issue #349

### CODE REVIEW

File: backend/apis/leads/update-lead.api.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        fullLead.lastName,
+        fullLead.age,
+        fullLead.gender,
+        userId,
+        leadId
+      );
+    }
+
+    if (queries.length === 0) {
+      console.log("No queries");
```
Comment (joeldenning) [inline backend/apis/leads/update-lead.api.js]: Please remove all the console logs throughout the PR

```suggestion
```

File: backend/apis/leads/update-lead.api.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+      res.send({
```
Comment (joeldenning) [inline backend/apis/leads/update-lead.api.js]: The PATCH api returns just the lead - no wrapper `{"lead": {}}` object.

```js
    res.send(oldLead)
```

File: frontend/view-edit-lead/edit-lead-inputs/basic-lead-information-inputs.component.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import { capitalize } from "lodash-es";
+
+export default function BasicLeadInformationInputs(
+  props: BasicLeadInforamtionInputsProps
```
Comment (joeldenning) [inline frontend/view-edit-lead/edit-lead-inputs/basic-lead-information-inputs.component.tsx]:
```suggestion
  props: BasicLeadInformationInputsProps
```

File: frontend/view-edit-lead/lead-home/view-edit-basic-lead-info.component.tsx [hunk 4]
```diff
@@ -1,34 +1,122 @@
...
+      const abortController = new AbortController();
+      easyFetch(`/api/leads/${props.lead.id}`, {
+        method: "PATCH",
+        body: apiStatus.newLeadData,
+        signal: abortController.signal
+      }).then(data => {
+        dispatchApiStatus({ type: "reset" });
+        props.leadUpdated(data.lead);
+        setEditing(false);
+      });
```
Comment (joeldenning) [inline frontend/view-edit-lead/lead-home/view-edit-basic-lead-info.component.tsx]:
```suggestion
      }).catch(err => {
        setTimeout(() => {
          throw err;
        })
      })
```

Comment (MeghanBomberger) [reply frontend/view-edit-lead/lead-home/view-edit-basic-lead-info.component.tsx]: Do we need this in the /view-edit-client/client-home/view-edit-basic-info.component.tsx file's easyFetch? The .catch is missing there.

**Pull request merged**